### PR TITLE
[Dropbox] Implement Indexing Feature

### LINKED ISSUE #1004: Dropbox: Indexing Feature
Implement Indexing Feature for Dropbox connector

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Closes #1004 

Added Document Sync Functionality for Dropbox connector.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

Note: The current implementation for document sync is slow in terms of performance. So, we're trying to use low level APIs instead of SDK. However, the approach will still remain the same.

### CODE REVIEW

File: connectors/sources/dropbox.py [hunk 1]
```diff
@@ -5,22 +5,38 @@
...
-from connectors.utils import CancellableSleeps, iso_utc
+from connectors.utils import (
+    TIKA_SUPPORTED_FILETYPES,
+    CancellableSleeps,
+    convert_to_b64,
+    iso_utc,
+)
 
 RETRY_COUNT = 3
+FILE_SIZE_LIMIT = 10485760  # Size in Bytes
```
Comment (vidok) [inline connectors/sources/dropbox.py:31]: How did you come up with 10MB limit?

File: connectors/sources/dropbox.py [hunk 2]
```diff
@@ -164,8 +250,130 @@ async def ping(self):
...
+            return
+
+        logger.debug(f"Downloading {attachment_name}")
+
+        document = {
+            "_id": attachment.id,
+            "_timestamp": attachment.server_modified,
+        }
+        temp_filename = ""
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
```
Comment (vidok) [inline connectors/sources/dropbox.py:300]: `get_content` is doing too much from my point of view. I think this block of code can be defined in its own function.

File: connectors/sources/dropbox.py [hunk 3]
```diff
@@ -164,8 +250,130 @@ async def ping(self):
...
+                _, data = self.dropbox_client.download_paper_files(  # pyright: ignore
+                    path=attachment.path_display
+                )
+                await async_buffer.write(data.content)
+            else:
+                logger.info(f"Skipping the download of a file: {attachment_name}")
+
+            temp_filename = str(async_buffer.name)
+
+        logger.debug(f"Calling convert_to_b64 for file : {attachment_name}")
```
Comment (vidok) [inline connectors/sources/dropbox.py:318]: base64 encoding can be also moved away from this function.

File: connectors/sources/dropbox.py (hunk 1 above)
Comment (moxarth-elastic) [reply connectors/sources/dropbox.py:31]: The ElasticSearch is having the default value for indexing a file of 10 MB and so we've kept this as a constant.

File: connectors/sources/dropbox.py (hunk 2 above)
Comment (moxarth-elastic) [reply connectors/sources/dropbox.py:300]: Yes, this block of code can be moved to another function.

**Review by vidok — APPROVED** [review]

**Pull request merged**